src/common/components/game/GameArea.js

```diff
@@ -0,0 +1,282 @@
+import React, { Component } from 'react';
+import { arrayOf, bool, func, number, object, string } from 'prop-types';
+import Paper from 'material-ui/Paper';
+import screenfull from 'screenfull';
+
+import Chat from '../multiplayer/Chat';
+import { inBrowser } from '../../util/browser';
+
+import Board from './Board';
+import EndTurnButton from './EndTurnButton';
+import EventAnimation from './EventAnimation';
+import ForfeitButton from './ForfeitButton';
+import FullscreenMessage from './FullscreenMessage';
+import FullscreenToggle from './FullscreenToggle';
+import GameNotification from './GameNotification';
+import PlayerArea from './PlayerArea';
+import Sfx from './Sfx';
+import SoundToggle from './SoundToggle';
+import Status from './Status';
+import Timer from './Timer';
+import TutorialTooltip from './TutorialTooltip';
+import VictoryScreen from './VictoryScreen';
+
+/* Props shared by GameArea and GameAreaContainer. */
+export const gameProps = {
+  player: string,
+  currentTurn: string,
+  usernames: object,
+  winner: string,
+
+  selectedTile: string,
+  playingCardType: number,
+
+  status: object,
+  target: object,
+  attack: object,
+
+  blueHand: arrayOf(object),
+  orangeHand: arrayOf(object),
+
+  bluePieces: object,
+  orangePieces: object,
+
+  blueEnergy: object,
+  orangeEnergy: object,
+
+  blueDeck: arrayOf(object),
+  orangeDeck: arrayOf(object),
+
+  blueDiscardPile: arrayOf(object),
+  orangeDiscardPile: arrayOf(object),
+
+  eventQueue: arrayOf(object),
+  sfxQueue: arrayOf(string),
+  tutorialStep: object,
+
+  history: object,
+  location: object,
+  match: object,
+
+  gameOver: bool,
+  isTutorial: bool,
+  isMyTurn: bool,
+  isSpectator: bool,
+  isAttackHappening: bool,
+
+  actionLog: arrayOf(object),
+  socket: object
+};
+
+export default class GameArea extends Component {
+  /* eslint-disable react/no-unused-prop-types */
+  static propTypes = {
+    ...gameProps,
+
+    chatOpen: bool,
+    message: string,
+
+    onPassTurn: func,
+    onForfeit: func,
+    onTutorialStep: func,
+    onSendChatMessage: func,
+    onActivateObject: func,
+    onClickGameArea: func,
+    onClickEndGame: func,
+    onNextTutorialStep: func,
+    onPrevTutorialStep: func,
+    onSelectTile: func
+  };
+  /* eslint-enable react/no-unused-prop-types */
+
+  state = {
+    areaHeight: 1250,
+    boardSize: 1000,
+    chatOpen: true
+  };
+
+  componentDidMount() {
+    this.updateDimensions();
+    window.addEventListener('resize', () => { this.updateDimensions(); });
+  }
+
+  handleToggleFullScreen = () => {
+    screenfull.toggle(this.gameArea);
+  };
+
+  handleToggleChat = () => {
+    this.setState(state => ({ chatOpen: !state.chatOpen }));
+  };
+
+  loadBackground = () => inBrowser() ? require('../img/black_bg_lodyas.png') : '';
+
+  updateDimensions = () => {
+    const maxBoardHeight = window.innerHeight - 64 - 150;
+    const maxBoardWidth = window.innerWidth - 256;
```

Comment:
Seems like we use constants like `64px` and `256px` a lot. Might be useful to have them as `HEADER_HEIGHT` and `SIDEBAR_WIDTH` or something.